elide-spring/elide-spring-boot-autoconfigure/README.md

```diff
@@ -0,0 +1,106 @@
+
+## Entity Dictionary Override
+
+By default, auto configuration creates an `EntityDictionary` with no checks or life cycle hooks registered. It does register spring as the dependency injection framework for Elide model injection.
+
+```java
+    @Bean
+    @ConditionalOnMissingBean
+    public EntityDictionary buildDictionary(AutowireCapableBeanFactory beanFactory) {
+        return new EntityDictionary(new HashMap<>(), beanFactory::autowireBean);
+    }
+```
+
+A typical override would add some checks and life cycle hooks.  *This is likely the only override you'll need*:
+
+```java
+    @Bean
+    @ConditionalOnMissingBean
+    public EntityDictionary buildDictionary(AutowireCapableBeanFactory beanFactory) {
+        HashMap<String, Class<? extends Check>> checkMappings = new HashMap<>();
+        checkMappings.put("allow all", Role.ALL.class);
+        checkMappings.put("deny all", Role.NONE.class);
+
+        EntityDictionary dictionary = new EntityDictionary(checkMappings, beanFactory::autowireBean);
+        dictionary.bindTrigger(Book.class, OnCreatePostCommit.class, (book, scope, changes) -> { /* DO SOMETHING */ }); 
+        dictionary.bindTrigger(Book.class, OnUpdatePostCommit.class, "title", (book, scope, changes) -> { /* DO SOMETHING */ });
+    }
+```
+
+## Data Store Override
+By default, the auto configuration will wire up a JPA data store:
+
+```java
+    @Bean
+    @ConditionalOnMissingBean
+    public DataStore buildDataStore(EntityManagerFactory entityManagerFactory) {
+        return new JpaDataStore(
+                () -> { return entityManagerFactory.createEntityManager(); },
+                (em -> { return new NonJtaTransaction(em); }));
+    }
+```
+
+Override this bean if you want a different store or multiple stores.
+
+## Swagger Override
+
+By default, Elide will generate swagger documentation for every model exposed into a single swagger document:
+
+```java
+    @Bean
+    @ConditionalOnMissingBean
+    public Swagger buildSwagger(EntityDictionary dictionary, ElideConfigProperties settings) {
+        Info info = new Info()
+                .title(settings.getSwagger().getName())
+                .version(settings.getSwagger().getVersion());
+
+        SwaggerBuilder builder = new SwaggerBuilder(dictionary, info);
+
+        Swagger swagger = builder.build().basePath(settings.getJsonApi().getPath());
+
+        return swagger;
+    }
+```
+
+You'll want to override this if:
+1. You want to configure authentication for your swagger endpoint.
+2. You don't want to expose all your models via swagger.
+3. You want to break up your models into multiple swagger documents.
+
+The Swagger controller will also except a bean that returns: `Map<String, Swagger`. This will break the swagger document into multiple documents.  They key of the map is the URL prefix for each separate document exposed.
+
+## Elide Override
+
+This provides complete control to manipulate the `ElideSettings` object that configures Elide.
```

Comment:
Is this referring to overriding `initializeElide()`? The other overrides have an example, we show add this for clarity.
```
@Bean
@ConditionalOnMissingBean
public Elide initializeElide(EntityDictionary dictionary,
    DataStore dataStore, ElideConfigProperties settings) {
        ....
```

Reply:
:+1: